codebase2/codebase/U/Codebase/Referent.hs

```diff
@@ -54,6 +54,11 @@ toReference = \case
   Ref termRef -> termRef
   Con typeRef _ -> typeRef
 
+toTermReference :: Referent' termRef typeRef -> Maybe termRef
```

Comment:
This behaves differently than `Referent.toReference` which I think is likely to be a footgun in the future;
https://github.com/unisonweb/unison/blob/trunk/unison-core/src/Unison/Referent'.hs#L57-L60

Not saying the behaviour of `Referent.toReference` is necessarily correct, but we definitely depend on its current behaviour a lot in  the `Backend` module right now 😬 

The fact that this returns a Maybe is hopefully enough of a signifier about what it's doing, but it still makes me a little nervous to have differing behaviour there.